client/src/util/csvExport.ts

```diff
@@ -0,0 +1,192 @@
+import AnnoMatrix from "../annoMatrix/annoMatrix";
+import { Genesets } from "../reducers/genesets";
+import {
+  STANDARD_CATEGORY_NAMES,
+  EXCLUDED_CATEGORY_NAMES,
+} from "../common/types/entities";
+import { Schema, Field } from "../common/types/schema";
+import { isCategoricalAnnotation } from "./stateManager/annotationsHelpers";
+
+/**
+ * Downloads a file by creating a blob and triggering a download
+ */
+export function downloadFile(content: string, filename: string): void {
+  const blob = new Blob([content], { type: "text/csv;charset=utf-8;" });
+  const url = URL.createObjectURL(blob);
+  const a = document.createElement("a");
+  a.href = url;
+  a.download = filename;
+  a.style.display = "none";
+  document.body.append(a);
+  a.click();
+  // Revoke the blob URL and remove the element
+  setTimeout(() => {
+    URL.revokeObjectURL(url);
+    a.remove();
+  }, 1000);
+}
+
+/**
+ * Escapes CSV values to handle commas, quotes, and newlines
+ */
+function escapeCsvValue(value: any): string {
```

Comment:
We don't use any typing in this codebase. I would suggest setting value to be string | number